File: Aac-data-module-coroutines/src/main/java/com/aac/data/coroutines/ResultJava.java
package com.aac.data.coroutines;

import android.support.annotation.Nullable;

/**
 * author  yangc
 * date 2018/3/15
 * E-Mail:yangchaojiang@outlook.com
 * Deprecated:
 */

public class ResultJava<T> {

    private  T result  ;
    private Throwable error ;

    public T getResult() {
        return result;
    }

    @Nullable
    public Throwable getError() {
        return error;
    }
    public static <T> ResultJava<T> of(Throwable error){
        ResultJava<T> bean=new ResultJava<>();
        bean.error=error;
        return bean;
    }
    public static <T> ResultJava<T> of(T bean){
        ResultJava<T> bean2=new ResultJava<>();
        bean2.result=bean;
        return bean2;
    }

    public T component1 () {  return result ; }
    public Throwable component2 () {  return error ; }
}

File: Aac-data-module-coroutines/src/main/java/com/aac/data/coroutines/ResultJava.kt
package com.aac.data.coroutines

/**
 * author  yangc
 * date 2018/3/15
 * E-Mail:yangchaojiang@outlook.com
 * Deprecated:
 */

class ResultJava<T> {

    var result: T? = null
        private set
    var error: Throwable? = null
        private set

    operator fun component1(): T? {
        return result
    }

    operator fun component2(): Throwable? {
        return error
    }

    companion object {
        fun <T> of(error: Throwable): ResultJava<T> {
            val bean = ResultJava<T>()
            bean.error = error
            return bean
        }

        fun <T> of(bean: T): ResultJava<T> {
            val bean2 = ResultJava<T>()
            bean2.result = bean
            return bean2
        }
    }
}
